substrate/frame/examples/basic/src/lib.rs

```diff
@@ -177,267 +188,199 @@ pub mod pallet {
 			Weight::zero()
 		}
 
-		// `on_finalize` is executed at the end of block after all extrinsic are dispatched.
+		// `on_finalize` is executed at the end of block after all extrinsics are dispatched.
 		fn on_finalize(_n: BlockNumberFor<T>) {
 			// Perform necessary data/state clean up here.
 		}
+	}
 
-		// A runtime code run after every block and have access to extended set of APIs.
-		//
-		// For instance you can generate extrinsics for the upcoming produced block.
-		fn offchain_worker(_n: BlockNumberFor<T>) {
-			// We don't do anything here.
-			// but we could dispatch extrinsic (transaction/unsigned/inherent) using
-			// sp_io::submit_extrinsic.
-			// To see example on offchain worker, please refer to example-offchain-worker pallet
-			// accompanied in this repository.
+	/// An example storage item to store a single value, in our case, some Balance.
+	/// This storage item uses [`OptionQuery`] by default which will return what is in the actual DB
+	/// state. If this is `None`, then the resulting query will be `None`.
+	///
+	/// The getter attribute generates a function on the `Pallet` struct that we can use to
+	/// conveniently retrieve the current value stored.
+	#[pallet::storage]
+	#[pallet::getter(fn dummy)]
+	pub(super) type Dummy<T: Config> = StorageValue<_, T::Balance>;
+
+	/// An example storage item that stores a `u32` value.
+	/// Here, we demonstrate using [`ValueQuery`] instead of the default query which will always
+	/// return a value.
+	///
+	/// In our example, if ever the stored value is removed, the value stored will be `0` which is
```

Comment:
This is a good one. What you can also do here is to demonstrate this in a test, and docify it here. Though I stil think this should go mostly in the documentation of `ValueQuery` itself so if anyone hovers into it, they see the good example.